src/transformers/modeling_tf_utils.py

```diff
@@ -1116,14 +1120,15 @@ def _get_resized_lm_head_decoder(self, old_lm_head_decoder, new_num_tokens):
             self._get_word_embedding_weight(self.get_input_embeddings()) == old_lm_head_decoder
         )
 
-        if old_lm_head_decoder is not None and not is_input_output_equals:
+        if old_lm_head_decoder is not None and (not is_input_output_equals or not self.config.tie_word_embeddings):
```

Comment:
I don't think `is_input_output_equals` is enough to decide whether the input and outputs are tied or not -> it's mostly the `tie_word_embeddings` attribute that does that